core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -174,4 +173,45 @@ object Runtime {
     val environment = r
     val platform    = platform0
   }
+
+  lazy val default = Runtime((), Platform.default)
+
+  lazy val global = Runtime((), Platform.global)
+
+  /**
+   * Builds a new runtime given an effectual and resourceful environment `R`
+   * and the default platform. This method is effectful because it executes
+   * acquiring the environment. Returns a finalizer that can be used to release
+   * the environment.
+   */
+  def unsafeDefault[R <: Has[_]](layer: ZLayer.NoDeps[Nothing, R]): (Runtime[R], Exit[Any, Any] => UIO[Any]) =
+    unsafeMake(layer, default.platform)
+
+  /**
+   * Builds a new runtime given an effectual and resourceful environment `R`
+   * and Scala's global execution context. This method is effectful because
+   * it executes acquiring the environment. Returns a finalizer that can be
+   * used to release the environment.
+   */
+  def unsafeGlobal[R <: Has[_]](layer: ZLayer.NoDeps[Nothing, R]): (Runtime[R], Exit[Any, Any] => UIO[Any]) =
+    unsafeMake(layer, global.platform)
+
+  /**
+   * Builds a new runtime given an effectual and resourceful environment `R`
+   * and a [[zio.internal.Platform]]. This method is effectful because it
+   * executes acquiring the environment. Returns a finalizer that can be used
+   * to release the environment.
+   */
+  def unsafeMake[R <: Has[_]](
+    layer: ZLayer.NoDeps[Nothing, R],
+    platform: Platform
+  ): (Runtime[R], Exit[Any, Any] => UIO[Any]) = {
```

Comment:
I think that while fully general-purpose, this is too confusing for general consumption (indeed, even I had to think carefully about what `Exit` would be reasonable to pass to the finalizer?). A user can always do this manually if they are "advanced", so this solution should probably be targeted toward mainstream use cases.

I'd probably do it like this:

```scala
/**
 * Unsafely creates a `Runtime` from a `ZLayer` whose resources will be allocated
 * immediately, and not released until the end of the application.
 * 
 * This method is useful for small applications and integrating ZIO with legacy code, but
 * other applications should investigate using `ZIO#provideLayer` directly in their 
 * application entry points.
 */
def unsafeGlobal[R <: Has[_]](layer: ZLayer.NoDeps[Any, R]): Runtime[R] = ???
```

Then if you add to `internal.PlatformSpecific.scala` something like `def addShutdownHook(f: () => Unit): Unit` that does something differently on JVM / JS, and use that inside the implementation of this method to call the finalizer on shutdown.